### PR TITLE
Support to show remaining logs with timeout if the batch failed

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_

If the batch failed, we shall try to show the remaining logs with timeout.

### _How was this patch tested?_
Existing UT.

### CODE REVIEW

**Comment by turboFei** [comment]
cc @lightning-L  @pan3793

File: kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/cmd/log/LogBatchCommand.scala [hunk 1]
```diff
@@ -45,43 +45,81 @@ class LogBatchCommand(
...
+                  size)
+                from += log.getLogRowSet.size
+                log.getLogRowSet.asScala.foreach(x => info(x))
+              } catch {
+                case e: Exception => error(s"Error fetching batch logs: ${e.getMessage}")
+              }
+
+              // if it has been done and the last fetched logs is non empty, do not wait a interval
+              if (!done || (log != null && log.getRowCount > 0)) {
+                Thread.sleep(conf.get(CTL_BATCH_LOG_QUERY_INTERVAL))
```
Comment (pan3793) [inline kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/cmd/log/LogBatchCommand.scala:88]: Should InterruptedException be caught here?

File: kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/cmd/log/LogBatchCommand.scala [hunk 2]
```diff
@@ -100,7 +138,7 @@ class LogBatchCommand(
     var batch: Batch = null
 
     if (!normalizedCliConfig.logOpts.forward) {
-      return (batch, true)
+      return (batch, log != null)
```
Comment (pan3793) [inline kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/cmd/log/LogBatchCommand.scala:141]: would you please explain a little about why this change is required?

Comment (turboFei) [reply kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/cmd/log/LogBatchCommand.scala:141]: before, the fetch log and check state in the same context.

Now the fetch log operation is in log thread.

File: kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/cmd/log/LogBatchCommand.scala [hunk 3]
```diff
@@ -79,7 +79,33 @@ class LogBatchCommand(
...
+          val timeout = conf.get(CTL_BATCH_LOG_ON_FAILURE_TIMEOUT)
+          var hasRemainingLogs = true
+          while (hasRemainingLogs && System.currentTimeMillis() - startTime < timeout) {
+            try {
+              val log = kyuubiInstanceBatchRestApi.getBatchLocalLog(
+                batchId,
+                from,
+                size)
+              from += log.getLogRowSet.size
+              log.getLogRowSet.asScala.foreach(x => info(x))
```
Comment (pan3793) [inline kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/cmd/log/LogBatchCommand.scala:99]: how about extracting the duplicated code block?

**Review by pan3793 — APPROVED** [review]
LGTM except the code dup

**Comment by turboFei** [comment]
> LGTM except the code dup

thanks, addressed.

File: kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/cmd/log/LogBatchCommand.scala [hunk 4]
```diff
@@ -55,15 +55,17 @@ class LogBatchCommand(
 
       withKyuubiInstanceRestClient(kyuubiRestClient, kyuubiInstance) { kyuubiInstanceRestClient =>
         val kyuubiInstanceBatchRestApi: BatchRestApi = new BatchRestApi(kyuubiInstanceRestClient)
+
+        def getOperationLogOnce(): OperationLog = {
```
Comment (pan3793) [inline kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/cmd/log/LogBatchCommand.scala:59]: prefer to use "fetch" or "retrieve" to distinguish w/ "getter", and it not fetch but also print the log, the method name should mention that, or introduce a param to control the display behavior.

**Comment by turboFei** [comment]
thanks, merging to master
